src/libsyntax/parse/parser.rs

```diff
@@ -3798,6 +3800,21 @@ impl<'a> Parser<'a> {
                                                             Vec<P<TypeBinding>>)> {
         let lifetimes = try!(self.parse_lifetimes(token::Comma));
 
+        // Fix #20616
+        if !lifetimes.is_empty() && self.last_token != Some(box token::Comma) {
+            match self.token {
+                token::Gt | token::Ge |
+                token::BinOpEq(token::Shr) | token::BinOp(token::Shr) => {}
```

Comment:
Is this really the best way to distinguish whether the last token started with >`?  I'll go along with it; just seems like a real footgun in general